File: Leetcode/1432 - Max Difference You Can Get From Changing an Integer/solution.ts
function maxDiff(num: number): number {
    const strNum: string = num.toString();

    const getMaxNum = (numStr: string): number => {
        let remapDig: string = '';
        const remapToDig: string = '9';
        for (const ch of numStr) {
            if (ch !== '9') {
                remapDig = ch;
                break;
            }
        }

        if (remapDig === '') return parseInt(numStr);
        const newStr: string = numStr.split('').map((dig: string) => dig === remapDig ? remapToDig : dig).join('');
        return parseInt(newStr);
    };

    const getMinNum = (numStr: string): number => {
        let remapDig: string = '';
        let remapToDig: string = '';

        if (numStr[0] !== '1') {
            remapDig = numStr[0];
            remapToDig = '1';
        } else {
            for (let i = 1; i < numStr.length; i++) {
                const dig: string = numStr[i];
                if (dig !== '0' && dig !== numStr[0]) {
                    remapDig = dig;
                    remapToDig = '0';
                    break;
                }
            }
        }

        if (remapDig === '') return parseInt(numStr);
        const newStr: string = numStr.split('').map((dig: string) => dig === remapDig ? remapToDig : dig).join('');
        return parseInt(newStr);
    };

    return getMaxNum(strNum) - getMinNum(strNum);
};

File: Leetcode/1432 - Max Difference You Can Get From Changing an Integer/solution.py
def maxDiff(self, num: int) -> int:
    num_str: str = str(num)

    def get_max_num(num_str: str) -> int:
        digit_to_replace: str = ''
        replacement_digit: str = '9'

        for ch in num_str:
            if ch != '9':
                digit_to_replace = ch
                break

        if digit_to_replace == '':
            return int(num_str)

        new_str: str = ''.join(
            replacement_digit if ch == digit_to_replace else ch
            for ch in num_str
        )
        return int(new_str)

    def get_min_num(num_str: str) -> int:
        digit_to_replace: str = ''
        replacement_digit: str = ''

        if num_str[0] != '1':
            digit_to_replace = num_str[0]
            replacement_digit = '1'
        else:
            for ch in num_str[1:]:
                if ch != '0' and ch != num_str[0]:
                    digit_to_replace = ch
                    replacement_digit = '0'
                    break

        if digit_to_replace == '':
            return int(num_str)

        new_str: str = ''.join(
            replacement_digit if ch == digit_to_replace else ch
            for ch in num_str
        )
        return int(new_str)

    return get_max_num(num_str) - get_min_num(num_str)